Aquarius/Src/Aquarius/Renderer/Texture.cpp

```diff
@@ -49,6 +53,32 @@ namespace Aquarius {
 		return success;
 	}
 
+	bool Texture::generateSolidTexture(std::array<uint8_t, 4> color)
+	{
+		glGenTextures(1, &m_ID);
+		glBindTexture(GL_TEXTURE_2D, m_ID);
+
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, static_cast<int>(m_Configuration.textureWrapS));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, static_cast<int>(m_Configuration.textureWrapT));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, static_cast<int>(m_Configuration.minTextureFiltering));
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, static_cast<int>(m_Configuration.magTextureFiltering));
+
+		m_Width = 1;
+		m_Height = 1;
+
+		uint8_t data[4];
+		memcpy(data, &color, sizeof(color));
```

Comment:
Is this because glTexImage2D requires a pointer to the data? If so, could instead do:
```C++
glTexImage2D(GL_TEXTURE_2D, 0, internalFormat, m_Width, m_Height, 0, format, GL_UNSIGNED_BYTE, color.data());
```
std::array::data returns a pointer to the first element

Reply:
Yeah this was weird, I first wasn't sure if you could make a texture from a single pixel, so i needed to copy the contents of color into an array of like 16 for 4 pixels (2x2 texture)

However, now that I know I need one pixel only I can just take the single pixel passed and create a texture from that - good call